File: src/prod/src/ktllogger/sys/ktlshim/driver/logger.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------


#include "logger.h"

NTSTATUS
KtlLogInitializeKtl()
{
    NTSTATUS status;
    
    KtlSystem* underlyingSystem;
    status = KtlSystem::Initialize(FALSE,             // Do not initialize VNetworking
                                   &underlyingSystem);

    if (!NT_SUCCESS(status)) {
        return status;
    }

    //
    // Turn on strict allocation tracking
    //
    underlyingSystem->SetStrictAllocationChecks(TRUE);

    EventRegisterMicrosoft_Windows_KTL();

    //
    // By default, enable debug trace mirroring.
    //
    KDbgMirrorToDebugger = FALSE;
    
    return(STATUS_SUCCESS);
}


VOID
KtlLogDeinitializeKtl()
{
    EventUnregisterMicrosoft_Windows_KTL();

    //
    // Reenable printfs in case there is some memory leaks.
    //
    KDbgMirrorToDebugger = TRUE;    
    KtlSystem::Shutdown();
}

File: src/prod/src/managed/infrastructurewrapper/wrappercommon/Logger.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace Microsoft.Fabric.InfrastructureWrapper
{
    using System;
    using System.Globalization;

    internal class Logger
    {
        public static void LogInfo(string type, string format, params object[] args)
        {
            var outputFormat = string.Format(CultureInfo.InvariantCulture, "Info: {0}:- {1}", type, format);
            Console.WriteLine(outputFormat, args);
        }

        public static void LogError(string type, string format, params object[] args)
        {
            var outputFormat = string.Format(CultureInfo.InvariantCulture, "Error: {0}:- {1}", type, format);
            Console.WriteLine(outputFormat, args);
        }

        public static void LogWarning(string type, string format, params object[] args)
        {
            var outputFormat = string.Format(CultureInfo.InvariantCulture, "Warning: {0}:- {1}", type, format);
            Console.WriteLine(outputFormat, args);
        }

        public static void LogFatalException(string type, Exception e)
        {
            Console.WriteLine("Exception: {0}:- {1}", type, e);
        }
    }
}